src/borg/remote.py

```diff
@@ -204,12 +204,23 @@ def serve(self):
                             else:
                                 logging.debug('\n'.join(ex_full))
 
-                            os.write(stdout_fd, msgpack.packb({MSGID: msgid,
-                                                               b'exception_class': e.__class__.__name__,
-                                                               b'exception_args': e.args,
-                                                               b'exception_full': ex_full,
-                                                               b'exception_short': ex_short,
-                                                               b'sysinfo': sysinfo()}))
+                            try:
+                                msg = msgpack.packb({MSGID: msgid,
+                                                    b'exception_class': e.__class__.__name__,
+                                                    b'exception_args': e.args,
+                                                    b'exception_full': ex_full,
+                                                    b'exception_short': ex_short,
+                                                    b'sysinfo': sysinfo()})
+                            except TypeError:
+                                msg = msgpack.packb({MSGID: msgid,
+                                                    b'exception_class': e.__class__.__name__,
+                                                    b'exception_args': [x if isinstance(x, (str, bytes, int)) else None
```

Comment:
I think for filtering on the first try the whitelist has to be better(i.e. support basic lists maybe) which would be more complicated code. Enkore’s first suggestion even was to just drop the args completely if they are troublesome. I’m not sure about the value of using repr instead of None, because these should end up in readable form in exception_full for human debugging anyway. exception_args is for safe automated processing and for recreating exceptions. Both not best served by repr·s.

Anyway this is only a fallback path, i don‘t think we have any exceptions that have information in non msgpackable types yet. And if we add new ones we might want to revisit this anyway.